packages/suite/src/utils/wallet/accountUtils.ts

```diff
@@ -292,6 +307,57 @@ export const enhanceTokens = (tokens: Account['tokens']) => {
         }));
 };
 
+export const enhanceAddresses = (
+    addresses: AccountAddresses | undefined,
+    networkType: Account['networkType'],
+    accountIndex: Account['index'],
+): AccountAddresses | undefined => {
+    // Addresses used in Suite include full derivation path including account index.
+    // These addresses are derived on a backend (Blockbook/Blockfrost) from a public key.
+    // In bitcoin an account index is encoded directly in a public key, so blockbook will extract it
+    // and return full derivation path for each derived address.
+    // (https://github.com/trezor/blockbook/blob/b82dc92522eee957b7a139c38269a1844fe102f8/bchain/coins/btc/bitcoinparser.go#L428)
+    // Since cardano account public key doesn't encode information about the account index, like Bitcoin does,
+    // Blockfrost backend returns partial derivation path where account index is replaced with character 'i'.
+    // So we rely on the client (this function) to replace it with correct account index.
+
+    if (!addresses) return undefined;
+    if (networkType !== 'cardano') return addresses;
+
+    const accountIndexStr = accountIndex.toString();
+    const used = addresses.used.map(address => ({
+        ...address,
+        path: address.path.replace('i', accountIndexStr),
+    }));
+    const unused = addresses.unused.map(address => ({
+        ...address,
+        path: address.path.replace('i', accountIndexStr),
+    }));
+    const change = addresses.change.map(address => ({
+        ...address,
+        path: address.path.replace('i', accountIndexStr),
+    }));
```

Comment:
nitpick: reduction of repetition

```suggestion
    const accountIndexStr = accountIndex.toString();
   
    const replaceAccountIndex = (address: AccountAddress) => ({
        ...address,
        path: address.path.replace('i', accountIndexStr),
    });

    const used = addresses.used.map(replaceAccountIndex);
    const unused = addresses.unused.map(replaceAccountIndex);
    const change = addresses.change.map(replaceAccountIndex);
```